sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,254 @@
+            var mockEventHubProperties = EventHubsModelFactory.EventHubProperties(mockEventHubName, fakeDate, new string[] { mockPartitionId });
+
+            // creating mocking partition properties
+            var mockPartitionProperties = EventHubsModelFactory.PartitionProperties(mockEventHubName, mockPartitionId, false, lastSequence, lastSequence, lastOffset, fakeDate);
+
+            // creating partition publishing properties
+            var mockPartitionPublishingProperties = EventHubsModelFactory.PartitionPublishingProperties(true, 675, (short)12, 4);
+
+            mockEventHubConsumerClient
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[] { mockPartitionId });
+
+            mockEventHubConsumerClient
+                .Setup(c => c.GetPartitionPropertiesAsync(mockPartitionId,It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockPartitionProperties);
+
+            mockEventHubConsumerClient
+                .Setup(c => c.ReadEventsFromPartitionAsync(It.IsAny<string>(), It.IsAny<EventPosition>(), It.IsAny<ReadEventOptions>(),It.IsAny<CancellationToken>()))
+                .Returns(GetMockEventsFromPartition(mockPartitionId,mockLastEnquedEventProperties))
+                .Verifiable();
+
+            var consumer = mockEventHubConsumerClient.Object;
+
+            try
+            {
+                using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+                string firstPartition = (await consumer.GetPartitionIdsAsync(cancellationSource.Token)).First();
+                PartitionProperties partitionProperties = await consumer.GetPartitionPropertiesAsync(firstPartition, cancellationSource.Token);
+                EventPosition startingPosition = EventPosition.FromOffset(partitionProperties.LastEnqueuedOffset);
+
+                var options = new ReadEventOptions
+                {
+                    TrackLastEnqueuedEventProperties = true
+                };
+
+                await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+                    firstPartition,
+                    startingPosition,
+                    options,
+                    cancellationSource.Token))
+                {
+                    LastEnqueuedEventProperties properties =
+                        partitionEvent.Partition.ReadLastEnqueuedEventProperties();
+
+                    Assert.AreEqual(mockPartitionId, partitionEvent.Partition.PartitionId);
+                    Assert.AreEqual(lastSequence, properties.SequenceNumber);
+                    Assert.AreEqual(lastOffset, properties.Offset);
+                    Assert.AreEqual(fakeDate, properties.EnqueuedTime);
+                    Assert.AreEqual(fakeDate, properties.LastReceivedTime);
+                }
```

Comment:
Should we assert the values here?

Reply:
We should probably do a "call the metadata method,  assert the values"  pattern,  interleaving the calls.   Something like:

```text
- Get Event Hub Properties
- Assert Event Hub Properties

- Get Partition Properties
- Assert Partition Properties
```

Reply:
On second thought, we can probably just show the query and then leave a comment for "the stuff you just read matches the stuff that you mocked"...

Reply:
removed the asserts and added comments

Reply:
please let me know if we should change the language of the comments :)

Reply:
I think the high level of them is great.   We'll eventually want to go with proper capitalization and punctuation to match the Event Hubs codebase.   Here, too, we'll need to double-check that our line lengths fit into the GitHub width to avoid scrolling.   

_(that's easier to do when looking at the markdown, so I suggest we leave it until that point)_